solution2.py

```diff
@@ -0,0 +1,42 @@
+# // Time Complexity : o(n*n + n/2*height => for loop inside addall at each leaf nodes) == o(n^2)
+# // Space Complexity : o(n*h)
+# // for complete binary tree, space complexity will be o(nlogn)
+# // else for a tree with single leaf with n nodes, o(n^2)
+# // Did this code successfully run on Leetcode : yes
+# // Any problem you faced while coding this : Not in particular
+#
+#
+# // Your code here along with comments explaining your approach
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def __init__(self):
+        self.sums = 0
+
+    def sumNumbers(self, root: TreeNode) -> int:
+        if not root:
+            return self.sums
+        r1 = []
```

Comment:
Please also try the other solution where a list path is not maintained. Much better Time & Space.